bin/bank/pycbc_brute_bank

```diff
@@ -247,20 +249,52 @@ class TriangleBank(object):
     def check_params(self, gen, params, threshold):
         num_tried = 0
         num_added = 0
-        for i in range(len(tuple(params.values())[0])):
-            num_tried += 1.0
-
-            try:
-                hp = gen.generate(**{key:params[key][i] for key in params})
-            except Exception as err:
-                print(err)
-                continue
-
-            hp.gen = gen
-            hp.threshold = threshold
-            if hp not in self:
-                num_added += 1
-                self.insert(hp)
+        total_num = len(tuple(params.values())[0])
+
+        if args.num_cores == 1:
+            for i in range(total_num):
+                num_tried += 1.0
+
+                try:
+                    hp = gen.generate(**{key:params[key][i] for key in params})
+                except Exception as err:
+                    print(err)
+                    continue
+
+                hp.gen = gen
+                hp.threshold = threshold
+                if hp not in self:
+                    num_added += 1
+                    self.insert(hp)
+
+        # use multiprocessing
+        elif args.num_cores > 1:
+            waveform_cache = []
+            with multiprocessing.Pool(args.num_cores) as pool:
+                for return_wf in pool.imap_unordered(
+                    wf_wrapper, 
+                    ({k: params[k][idx] for k in params} for idx in range(total_num))
+                ):
+                    waveform_cache += [return_wf]
+            
+            for i in range(total_num):
+                num_tried += 1
```

Comment:
`num_tried = total_num` or am I missing something?

Reply:
You're right. corrected